lib/install/management/upgrade.go

```diff
@@ -96,10 +114,8 @@ func (d *Dispatcher) Upgrade(vch *vm.VirtualMachine, conf *config.VirtualContain
 		return err
 	}
 
-	// do clean up aggressively, even the previous operation failed with context deadline excceeded.
-	d.ctx = context.Background()
-	d.cleanSnapshot(*snapshotRefID, snapshotName, conf.Name)
-	d.deleteUpgradeImages(ds, settings)
+	deleteSnapshot = true
```

Comment:
though with these assignments here, is there a scenario in which the deleteSnapshot and deleteImage lambdas would _not_ be called at the bottom here? Because if they're always called, which seems to be the case, it would make more sense to me to break them out into helper functions & call them here directly, instead of using `defer` and testing to see if `deleteSnapshot` & `deleteImage` are `true`, since they always seem to be `true` at the end of the method here